src/notification.c

```diff
@@ -495,14 +498,10 @@ void notification_init(struct notification *n)
                 default:
                         g_error("Unhandled urgency type: %d", n->urgency);
         }
-        if (!n->colors.fg)
-                n->colors.fg = g_strdup(defcolors.fg);
-        if (!n->colors.bg)
-                n->colors.bg = g_strdup(defcolors.bg);
-        if (!n->colors.highlight)
-                n->colors.highlight = g_strdup(defcolors.highlight);
-        if (!n->colors.frame)
-                n->colors.frame = g_strdup(defcolors.frame);
+        if (!COLOR_VALID(n->colors.fg)) n->colors.fg = defcolors.fg;
+        if (!COLOR_VALID(n->colors.bg)) n->colors.bg = defcolors.bg;
+        if (!COLOR_VALID(n->colors.highlight)) n->colors.highlight = defcolors.highlight;
+        if (!COLOR_VALID(n->colors.frame)) n->colors.frame = defcolors.frame;
```

Comment:
You're a bit inconsistent on your style (see `src/rules.c`). Sometimes you make single statement if a oneliner and sometimes it's put on two lines. Personally, I prefer to have the two line variant.

Reply:
Right. I try to minimize the use of lines, so if both the condition and statements are really short I put them in a single line (especially if there are a lot chained like here). I will try to be more consistent in the future